packages/mask/src/plugins/Tips/SNSAdaptor/TipsEntranceDialog.tsx

```diff
@@ -0,0 +1,359 @@
+}
+const useStyles = makeStyles()((theme) => ({
+    walletBtn: {
+        fontSize: '14px',
+    },
+    alertBox: {
+        marginBottom: '20px',
+    },
+    actions: {
+        position: 'absolute',
+        bottom: 16,
+        width: 'calc( 100% - 32px)',
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        gap: theme.spacing(1.5),
+    },
+    dContent: {
+        height: 600,
+        position: 'relative',
+        boxSizing: 'border-box',
+    },
+    btnContainer: {
+        width: '100%',
+        display: 'flex',
+        flexDirection: 'row-reverse',
+    },
+    loading: {
+        position: 'absolute',
+        top: '50%',
+        left: '50%',
+        transform: 'translate(-50%,-50%)',
+    },
+}))
+
+enum BodyViewStep {
+    Main = 'Tips',
+    Setting = 'Settings',
+    Wallets = 'Wallets',
+    AddWallet = 'Add Wallet',
+}
+
+interface WalletButtonProps extends ButtonProps {
+    step: BodyViewStep
+}
+
+const WalletButton: FC<WalletButtonProps> = ({ step, onClick }) => {
+    const { classes } = useStyles()
+    if (step === BodyViewStep.AddWallet) return null
+    return (
+        <div className={classes.btnContainer}>
+            <Button onClick={onClick} className={classes.walletBtn} variant="contained" size="small">
+                {step === BodyViewStep.Wallets ? BodyViewStep.AddWallet : BodyViewStep.Wallets}
+            </Button>
+        </div>
+    )
+}
+
+export function TipsEntranceDialog({ open, onClose }: TipsEntranceDialogProps) {
+    const { t } = useI18N()
+    const { classes } = useStyles()
+    const [showAlert, setShowAlert] = useState(true)
+    const [bodyViewStep, setBodyViewStep] = useState<BodyViewStep>(BodyViewStep.Main)
+    const [hasChanged, setHasChanged] = useState(false)
+    const [rawPatchData, setRawPatchData] = useState<BindingProof[]>([])
+    const [rawWalletList, setRawWalletList] = useState<BindingProof[]>([])
+    const supportedNetworks = useActivatedPlugin(PluginId.Tip, 'any')?.enableRequirement?.web3?.[
+        NetworkPluginID.PLUGIN_EVM
+    ]?.tipsSupportedChains
+    const { showSnackbar } = useCustomSnackbar()
+    const account = useAccount()
+    const nowTime = formatDateTime(new Date(), 'yyyy-MM-dd HH:mm')
+
+    const { value: currentPersonaIdentifier } = useAsyncRetry(() => {
+        setShowAlert(true)
+        return Services.Settings.getCurrentPersonaIdentifier()
+    }, [open])
+    const { value: currentPersona } = useAsyncRetry(
+        () => Services.Identity.queryPersona(currentPersonaIdentifier as ECKeyIdentifier),
+        [currentPersonaIdentifier],
+    )
+    const clickBack = () => {
+        if (bodyViewStep === BodyViewStep.Main) {
+            onClose()
+        } else {
+            setBodyViewStep(BodyViewStep.Main)
+        }
+    }
+    const { value: kv, retry: retryKv } = useKvGet()
+    const { loading, value: proofRes, retry: retryProof } = useProvedWallets()
+
+    useEffect(() => {
```

Comment:
Please refactor to use `useMemo`, and you can take to replace those local states.

```ts
const { rawWalletList, rawPatchData } = useMemo(() => {
    // computation
})
```